app/components/file_upload_preview/file_upload_preview.js

```diff
@@ -0,0 +1,156 @@
+        }).isRequired,
+        channelId: PropTypes.string.isRequired,
+        createPostRequestStatus: PropTypes.string.isRequired,
+        files: PropTypes.array.isRequired,
+        rootId: PropTypes.string,
+        uploadFileRequestStatus: PropTypes.string.isRequired,
+        uploadFileRequestId: PropTypes.number
+    };
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.createPostRequestStatus === RequestStatus.STARTED && nextProps.createPostRequestStatus === RequestStatus.SUCCESS) {
+            this.props.actions.handleClearFiles(this.props.channelId, this.props.rootId);
+        }
+    }
+
+    buildFilePreviews = () => {
+        return this.props.files.map((file) => {
+            return (
+                <View
+                    key={file.id}
+                    style={style.preview}
+                >
+                    <FileAttachmentPreview file={file}/>
+                    <TouchableOpacity
+                        style={style.removeButton}
+                        onPress={() => this.props.actions.handleRemoveFile(file.id, this.props.channelId, this.props.rootId)}
+                    >
+                        <Font
+                            name='times'
+                            color='#fff'
+                            size={15}
+                        />
+                    </TouchableOpacity>
+                </View>
+            );
+        });
+    }
+
+    handleCancelUpload = () => {
+        this.props.actions.cancelUploadFileRequest(this.props.uploadFileRequestId);
+    }
+
+    render() {
+        if (!this.props.files.length && this.props.uploadFileRequestStatus !== RequestStatus.STARTED) {
+            return null;
+        }
+
+        let component;
+        if (this.props.uploadFileRequestStatus === RequestStatus.STARTED) {
+            component = (
+                <View style={style.loader}>
+                    <Text style={style.loaderText}>{'Loading...'}</Text>
+                    <TouchableOpacity onPress={this.handleCancelUpload}>
+                        <Text style={style.cancelText}>{'Cancel'}</Text>
+                    </TouchableOpacity>
+                </View>
+            );
+        } else {
+            component = (
+                <ScrollView
+                    horizontal={true}
+                    style={style.scrollView}
+                    contentContainerStyle={style.scrollViewContent}
+                >
+                    {this.buildFilePreviews()}
+                </ScrollView>
+            );
+        }
+
+        return (
+            <KeyboardAvoidingView style={style.container}>
+                {component}
+            </KeyboardAvoidingView>
+        );
+    }
+}
+
+const style = StyleSheet.create({
+    cancelText: {
+        padding: 15,
+        color: '#fff',
+        fontSize: 18
+    },
+    container: {
+        backgroundColor: 'rgba(0, 0, 0, 0.5)',
+        bottom: 40,
+        height: deviceHeight,
```

Comment:
Does this still need to be addressed since we're disabling rotation? I was getting some strange behavior with `height: '100%'` to where it was cutting off the view and not actually taking up the entire view.